EIPS/eip-6051.md

```diff
@@ -0,0 +1,474 @@
+
+In addition to a `version` parameter, the following keys and functions are involved:
+
+1.  The Sender's private key  `sk`, which is to be encapsulated to the Recipient, and the corresponding address `account`.
+2.  The ephemeral Recipient key pair  `(r, R)`  such that `R = [r]G`. `G` denotes the base point of the elliptic curve, and `[r]G` denotes scalar multiplication. Optionally, `R` could be signed, and `signerPubKey` and `signature` are then provided for Sender to verify if `R`  could be trusted or not.
+3.  The ephemeral Sender key pair  `(s, S)`  such that `S = [s]G`.
+4.  The share secret  `ss := [s]R = [r]S` according to ECDH. Note that for secp256k1 this EIP follows RFC5903 and uses compact representation, which means to use *only* the `x` coordinate as the shared secret. For Curve25519 this EIP follows RFC7748.
+5.  The out-of-band data `oob`, optional. This could be digits or an alpha-numeric string entered by the user.
+6.  Let  `derivedKey  :=  HKDF(hash=SHA256, ikm=ss, info=oob,  salt,  length)`. HKDF is defined in RFC5869. The  `length`  should be determined by  `skey`  and  `IV`  requirements such that the symmetric key  `skey  = derivedKey[0:keySize]`, and  `IV  = derivedKey[keySize:length]`.  `keySize`  denotes the key size of the underlying symmetric algorithm, for example, 16 (bytes) for AES-128, and 32 (bytes) for Chacha20. See **Security Considerations** for the use of  `salt`.
+7.  Let `cipher  := authenticated_encryption(symAlg,  skey,  IV, data=sk)`. The symmetric cipher algorithm `symAlg` and authentication scheme are decided by the version parameter. No additional authentication data `aad` is used.
+
+A much-simplified example flow without signature and verification is:
+
+1.  *Recipient Application* generates  `(r, R)`.
+2.  User inputs  `R`  to *Sender Application*, along with a six-digit code “123456” as  `oob`.
+3.  *Sender Application* generates  `(s, S)`, and computes  `cipher`, then returns `S || cipher`.
+4.  *Recipient Application* scans to read  `S`  and  `cipher`. The user enters “123456” as  `oob`  to *Recipient Application*.
+5.  *Recipient Application* decrypts  `cipher`  to get  `sk`.
+6.  *Recipient Application* derives the address corresponding to  `sk`  so that the user can confirm the correctness.
+
+With signature and verification, the signature to `R` by `singerPubKey` is appended to `R`. `signerPubKey` itself could have been already signed by `trustedPubKey`, and that signature is appended to `signerPubKey`. Note that the signature is applied to the byte array data instead of its string representation, which might lead to confusion and interoperability issues (such as hex or base64, lower case v.s. upper case, etc.). See **Requests** and **Test Cases** for further clarification and examples.
```

Comment:
Why not link to them?
```suggestion
With signature and verification, the signature to `R` by `singerPubKey` is appended to `R`. `signerPubKey` itself could have been already signed by `trustedPubKey`, and that signature is appended to `signerPubKey`. Note that the signature is applied to the byte array data instead of its string representation, which might lead to confusion and interoperability issues (such as hex or base64, lower case v.s. upper case, etc.). See [Requests](#requests) and [Test Cases](#test-cases) for further clarification and examples.
```